src/GraphQL/GraphQLExtensions.cs

```diff
@@ -528,72 +579,26 @@ public static IValue AstFromValue(this object value, ISchema schema, IGraphType
                 {
                     var values = list
                         .Cast<object>()
-                        .Select(item => AstFromValue(item, schema, itemType))
+                        .Select(item => ToAST(itemType, item))
                         .ToList();
 
                     return new ListValue(values);
                 }
 
-                return AstFromValue(value, schema, itemType);
+                return ToAST(itemType, value);
             }
 
             // Populate the fields of the input object by creating ASTs from each value
             // in the dictionary according to the fields in the input type.
             if (type is IInputObjectGraphType input)
             {
-                if (!(value is Dictionary<string, object> dict))
-                {
-                    return null;
-                }
-
-                var fields = dict
-                    .Select(pair =>
-                    {
-                        var fieldType = input.GetField(pair.Key)?.ResolvedType;
-                        return new ObjectField(pair.Key, AstFromValue(pair.Value, schema, fieldType));
-                    })
-                    .ToList();
-
-                return new ObjectValue(fields);
+                return input.ToAst(value) ?? throw new InvalidOperationException($"Unable to convert the '{value}' of the object type '{input.Name}' to an AST representation.");
```

Comment:
The same problem here. See https://github.com/graphql-dotnet/graphql-dotnet/pull/2321/files#r584118949

Reply:
Also need to write a test with complex value.

Reply:
OK. Now I see that it's a call to instance method.

Reply:
So ok to resolve conversation?

Reply:
Yes. In general tests are required to verify complex default values.